### PR TITLE
[MRG] Depr preload_data

### LINKED ISSUE #2500: FIX: Deprecate raw.preload_data
For @teonlamont, to make sure we don't release 0.10 without it. From #2498.

### PR DESCRIPTION
**Pull request by teonlamont**
Closes #2500.

### CODE REVIEW

File: mne/io/base.py [hunk 1]
```diff
@@ -425,6 +425,30 @@ def _read_segment_file(self, data, idx, offset, fi, start, stop,
...
+            The raw object with data.
+
+        Notes
+        -----
+        This function will load raw data if it was not already preloaded.
+        If data were already preloaded, it will do nothing.
+        """
+        if not self.preload:
+            self._preload_data(True)
+        return self
```
Comment (Eric89GXL) [inline mne/io/base.py]: as with most deprecations, I would have just done `return self.load_data()`, but this will work :)

**Comment by Eric89GXL** [comment]
Will merge once the CIs come back happy (e.g., for style checks), even though I expect them to

**Comment by Eric89GXL** [comment]
Description edited to enable issue auto-close

**Comment by Eric89GXL** [comment]
Actually, can you update the `whats_new` API section?

File: mne/epochs.py [hunk 2]
```diff
@@ -40,7 +40,7 @@
 from .viz import (plot_epochs, _drop_log_stats,
                   plot_epochs_psd, plot_epochs_psd_topomap)
 from .utils import (check_fname, logger, verbose, _check_type_picks,
-                    _time_mask, check_random_state, object_hash)
+                    _time_mask, check_random_state, object_hash, deprecated)
```
Comment (Eric89GXL) [inline mne/epochs.py]: I was just about to tell you about this missing import :)

File: mne/io/base.py [hunk 3]
```diff
@@ -425,6 +425,30 @@ def _read_segment_file(self, data, idx, offset, fi, start, stop,
...
+            The raw object with data.
+
+        Notes
+        -----
+        This function will load raw data if it was not already preloaded.
+        If data were already preloaded, it will do nothing.
+        """
+        if not self.preload:
+            self._preload_data(True)
+        return self.load_data()
```
Comment (Eric89GXL) [inline mne/io/base.py]: I meant that the entire contents of the function could just be `return self.load_data()`. Usually when deprecating if possible you should just entirely wrap to the new function. As it stands now you're duplicating executed checks here and in `load_data`. The way you had it before, the code was not as DRY. doing `def preload_data(self): return self.load_data()` doesn't have either problem.

File: mne/io/base.py [hunk 4]
```diff
@@ -425,6 +425,30 @@ def _read_segment_file(self, data, idx, offset, fi, start, stop,
         raise NotImplementedError
 
     @verbose
+    @deprecated("This method has been renamed 'load_data' and will be removed "
+                "in v0.11.")
+    def preload_data(self, verbose=None):
```
Comment (Eric89GXL) [inline mne/io/base.py]: You'll also need to add this method name to the list of ignored things in `test_docstring_parameters`, it's why Travis is angry. Try running `nosetests mne/tests/test_docstring_parameters.py` and you'll see it.

File: mne/io/base.py (hunk 3 above)
Comment (agramfort) [inline mne/io/base.py]: yes just do self.load_data()

**Comment by Eric89GXL** [comment]
@teonlamont you'll need a rebase. You shouldn't need to manually specify the deprecated function in the list of ignores anymore.

**Comment by teonlamont** [comment]
ready for merge

File: mne/tests/test_docstring_parameters.py [hunk 5]
```diff
@@ -66,6 +66,15 @@ def get_name(func):
...
     'mne.gui.coregistration',  # deprecated single argument w/None
+    'mne.utils.plot_epochs_trellis',  # deprecated
+    'mne.utils.write_bem_surface',  # deprecated
+    'generate_sparse_stc',  # deprecated
+    'generate_stc',  # deprecated
+    'generate_evoked',  # deprecated
+    'generate_noise_evoked',  # deprecated
+    'mne.gui.coregistration',  # deprecated
+    'mne.utils.plot_topo',
+    'mne.viz.plot_image_epochs',  # deprecated
```
Comment (Eric89GXL) [inline mne/tests/test_docstring_parameters.py]: @teonlamont you shouldn't need any of these if you have rebased properly

**Comment by teonlamont** [comment]
fixed it

**Comment by Eric89GXL** [comment]
Thanks @teonlamont

**Pull request merged**